### ISSUE TITLE
Primary key migration from optional to required deleting rows

### ISSUE THREAD

**Issue by emili1996** [realm/realm-java#5899]
Hi guys, i just encountered a weird problem and don't know if is a bug or i just don't know how to
migrate things. 
I found a workaround for that, create new field with required attribute, transform to get the value of the old primary key, remove old primary key and set primary key to the newly created field.

#### Goal
Change primary key name and required to true (Kotlin non-null but not @Required)

#### Expected Results
The primary key with the new name and the field to be required

#### Actual Results
Migration removing 4 of 5 rows in the table after seting setRequired(Y, true)
I checked the operation before and all the rows have the primary key with a value and are not null

#### Steps & Code to Reproduce
Didn't check with another project but in my case was:
Have a realm object schema with nullable primary key.
Multiple objects of that type with values in the primary key.
Make a migration changing the name and required to true
The new table contains only one row and not the 5 before the migration

#### Code Sample

```java

> Your code here. Bigger samples should ideally be as separate Android Studio project, 
> in gists/repositories or privately at help@realm.io)

```

#### Version of Realm and tooling
Realm version(s): ?
4.3.3
Realm sync feature enabled: yes/no
No
Android Studio version: ?
3.1.1
Which Android version and device: ?
7.0

**Comment by Zhuinden** [comment]
Code sample.

**Comment by Zhuinden** [comment]
I'm kinda surprised that `renameField` actually worked.

> I found a workaround for that, create new field with required attribute, transform to get the value of the old primary key, remove old primary key and set primary key to the newly created field.

This is what I would have recommended, but the official Realm guys will have to look at what `renameField` does to a primary key field, I guess.

**Comment by cmelchior** [comment]
Hi @emili1996 
Thanks for the code. I'll see if I can reproduce it.

**Comment by emili1996** [comment]
No problem! Let me know if you need more code or background information.

**Comment by cmelchior** [comment]
@emili1996 I was able to reproduce your scenario, but I do not have a root cause yet.

**Comment by cmelchior** [comment]
Fixed in #5902
